src/basic/locale-util.c

```diff
@@ -303,6 +304,12 @@ bool is_locale_utf8(void) {
         } else if (r != -ENXIO)
                 log_debug_errno(r, "Failed to parse $SYSTEMD_UTF8, ignoring: %m");
 
+        /* This function may be called from libsystemd, and setlocale() is not thread safe. Assuming yes. */
```

Comment:
did you mean "Assuming no"?

Reply:
Gh, the comment is correct, but the code was wrong. Fixed.